--- src/TerrariaNetCore/Terraria/DelegateMethods.cs
+++ src/tModLoader/Terraria/DelegateMethods.cs
@@ -5,6 +_,7 @@
 using Terraria.Enums;
 using Terraria.Graphics.Shaders;
 using Terraria.ID;
+using Terraria.ModLoader;
 
 namespace Terraria;
 
@@ -477,7 +_,7 @@
 		if (tile2 == null)
 			return false;
 
-		if (tile2.type < 0 || tile2.type >= TileID.Count)
+		if (tile2.type < 0)
 			return false;
 
 		if (Main.tileSolid[tile2.type] && !TileID.Sets.Platforms[tile2.type])
@@ -723,7 +_,10 @@
 
 	public static bool NotDoorStand(int x, int y)
 	{
+		/*
 		if (Main.tile[x, y] != null && Main.tile[x, y].active() && Main.tile[x, y].type == 11) {
+		*/
+		if (Main.tile[x, y] != null && Main.tile[x, y].active() && TileLoader.CloseDoorID(Main.tile[x, y]) >= 0) {
 			if (Main.tile[x, y].frameX >= 18)
 				return Main.tile[x, y].frameX < 54;
 
